Lib/pathlib.py

```diff
@@ -1415,21 +1415,29 @@ def absolute(self):
         """
         if self.is_absolute():
             return self
+        elif self.root:
+            drive = os.path.splitroot(os.getcwd())[0]
+            return self._from_parsed_parts(drive, self.root, self._tail)
         elif self.drive:
```

Comment:
I'd find this more readable. The `cwd` variable is really set on one of two branches: line 1423 and line 1425. It makes sense for those two branches to be a separate `if/else` suite, after the first two early-return opportunites, I think:

```suggestion
        if self.root:
            drive = os.path.splitroot(os.getcwd())[0]
            return self._from_parsed_parts(drive, self.root, self._tail)
        if self.drive:
```